client/src/components/App.js

```diff
@@ -5,22 +5,71 @@ import NavHeader from "./NavHeader";
 import Creator from "./Creator";
 import Example from "./Example";
 import Fan from "./Fan";
+import Web3 from "web3";
+import Web3Modal from "web3modal";
+import Fortmatic from "fortmatic";
+
+let provider = null;
+let web3Modal = null;
+let web3 = null;
+let accounts = null;
+let mainAddress = "";
+
+const providerOptions = {
+  fortmatic: {
+    package: Fortmatic, // required
+    options: {
+      key: process.env.FORTMATIC_KEY, // required
+    },
+  },
+};
 
 class App extends Component {
   constructor(props) {
     super(props);
+    this.state = {
+      connectedWallet: "",
+    };
     this.openWalletConnectModal = this.openWalletConnectModal.bind(this);
   }
 
-  openWalletConnectModal() {
-    // TODO: Open web3modal and set connectedWallet when rendering Fan.
-    console.log("Opened wallet connect");
+  async openWalletConnectModal() {
+    if (!provider) {
+      web3Modal = new Web3Modal({
+        cacheProvider: true, // optional
+        providerOptions, // required
+      });
+      provider = await web3Modal.connect();
+      web3 = new Web3(provider);
```

Comment:
I'm assuming we no longer need the getWeb3 helper class now that we have this. Do you wanna remove it here or in a separate PR?